src/store.js

```diff
@@ -0,0 +1,88 @@
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Axios from 'axios'
+import { QUERRY } from './constant'
+Vue.use(Vuex)
+export const store = new Vuex.Store({
+  state: {
+    armyint: [],
+    DroidArmy: [],
+    TroopersArmy: [],
+    BattleResult: [],
+    BattleHistory: [],
+    count: null,
+    imgurl: '',
+    armydata: []
+  },
+  actions: {
+    getDroid ({ commit }) {
+      Axios
+        .get(QUERRY.GET_DROID)
+        .then(data => {
+          let post = data.data
+          commit('SET_DROID', post)
+        })
+        .catch(error => { console.log(error) })
```

Comment:
handle the errors gracefully. show a notification try this https://www.npmjs.com/package/vue-notification